server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -699,6 +701,25 @@ export class AssetService {
       throw new ForbiddenException();
     }
   }
+
+  async createAssetsSharedLink(authUser: AuthUserDto, dto: CreateAssetsShareLinkDto): Promise<SharedLinkResponseDto> {
+    const assets = [];
+
+    for (const assetId of dto.assetIds) {
+      const asset = await this._assetRepository.getById(assetId);
+      assets.push(asset);
+    }
```

Comment:
This is where we probably have some work to do still:

1. If an asset is not found, we should probably either throw a bad request or just skip it and then return that information.
2. We should definitely check the user has access to the asset they're trying to add.
3. I don't think we _neeeeeeed_ to load the asset (and I believe this loads all the child relations as well) just to save the id, (which we already have) to the shared link.

I'd test out doing something like this:

```typescript
const hasAccess = await this.accessChecker.checkAsset(...);
if(!hasAccess){
  throw BadRequestException();
}
// only the id is needed to create the association.
assets.push({ id: assetId } as AssetEntity); 
```